corebuild/integration/ILLink.Tasks/ILLink.CrossGen.targets

```diff
@@ -0,0 +1,231 @@
+         but the directory paths have to be passed with
+         platform-specific path separators in the crossgen command.
+         -->
+    <ItemGroup>
+      <_CrossGenPlatformAssemblies Include="$(CrossGenPlatformAssemblies)" />
+      <__CrossGenPlatformAssembliesPaths Include="@(_CrossGenPlatformAssemblies->'%(RootDir)%(Directory)')" />
+      <_CrossGenPlatformAssembliesPaths Include="@(__CrossGenPlatformAssembliesPaths->Distinct())" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <_PathSeparator>$([System.IO.Path]::PathSeparator)</_PathSeparator>
+      <_CrossGenPlatformAssembliesPaths>@(_CrossGenPlatformAssembliesPaths, '$(_PathSeparator)')</_CrossGenPlatformAssembliesPaths>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <CrossGenCommand>$(CrossGenExe) -readytorun -in $(CrossGenInput) -out $(CrossGenOutput) -platform_assemblies_paths $(_CrossGenPlatformAssembliesPaths)</CrossGenCommand>
+    </PropertyGroup>
+
+    <Message Text="$(CrossGenCommand)" />
+    <Exec Command="$(CrossGenCommand)" />
+
+  </Target>
+
+  <Target Name="_ComputeCrossGenInputs"
+          DependsOnTargets="_ComputeCrossGenPlatformAssemblies;_ComputeFilesToCrossGen" />
+
+  <!-- Compute the platform assembly paths, a parameter to crossgen
+       that lets it find dependencies of the input file. This needs to
+       include the path to the input file and its dependency closure,
+       including System.Private.CoreLib. -->
+  <Target Name="_ComputeCrossGenPlatformAssemblies"
+          DependsOnTargets="_ComputeManagedResolvedAssembliesForCrossGen">
+
+    <ItemGroup>
+      <_CrossGenPlatformAssemblies Include="@(_ManagedResolvedAssembliesForCrossGen)" />
+      <_CrossGenPlatformAssemblies Include="@(IntermediateAssembly)" />
+    </ItemGroup>
+
+  </Target>
+
+  <!-- _ManagedResolvedAssembliesForCrossGen includes the app's
+       managed dependency closure, including System.Private.CoreLib,
+       but not the app itself or resource assemblies. -->
+  <Target Name="_ComputeManagedResolvedAssembliesForCrossGen">
+    <ComputeManagedAssemblies Assemblies="@(ResolvedAssembliesToPublish)">
+      <Output TaskParameter="ManagedAssemblies" ItemName="_ManagedResolvedAssembliesForCrossGen" />
+    </ComputeManagedAssemblies>
+
+    <ItemGroup>
+      <_ManagedResolvedAssembliesForCrossGen Remove="@(_ManagedResolvedAssembliesForCrossGen->WithMetadataValue('AssetType', 'resources'))" />
+    </ItemGroup>
+  </Target>
+
+  <UsingTask TaskName="FilterByMetadata" AssemblyFile="$(LinkTaskDllPath)" />
+  <!-- This computes the default set of files that we want to be
+       crossgen'd. Some of these may already be R2R images, and these
+       will not be crossgen'd again. The default is to crossgen the
+       app and platform libraries. Defaults will be used only if
+       FilesToCrossGen hasn't been set elsewhere, allowing users and
+       other props/targets to select what will be crossgen'd. -->
```

Comment:
We need to ensure that when the linker part of the publish step, we crossgen  those and only those assemblies that were  crossgen-ed before linking. I don't think that default ensures that.